### ISSUE TITLE
allow TabularMSA `positional_metadata` to define second axis length

### ISSUE THREAD

**Issue by jairideout** [biocore/scikit-bio#1198]
@ebolyen and I discussed this: it'd be consistent to allow TabularMSA's second axis (`position`) to have a length even if the first axis has length 0 (i.e., no sequences).

Known changes to make this happen:

- `__init__`
- `__bool__`
- `shape`
- `extend`

There may be other parts of the code that assume zero sequences implies zero positions. It'll be important to check each method for this assumption.

Since this is a new border case we'll want to have unit tests added even for methods that weren't affected directly (since we're testing zero sequence/position cases).

**Comment by ebolyen** [comment]
It should be noted that this isn't a high priority as the edge cases are rare and weird anyhow.

**Comment by jairideout** [comment]
Agree, and it isn't possible to get into a buggy state currently, this would purely be an enhancement.

**Comment by ebolyen** [comment]
`__getitem__`
